samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/util/QuestionUtils.java

```diff
@@ -0,0 +1,38 @@
+package org.sakaiproject.tool.assessment.util;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class QuestionUtils {
+
+    /**
+     * Replace text variables for Calculated Questions
+     * @param text
+     * @param openSeparator
+     * @param closeSeparator
+     * @param replacement
+     * @return
+     */
+	public static String replaceTextVariables(String text, String openSeparator, String closeSeparator, String replacement) {
+		String result = text;
+		if (StringUtils.isNotBlank(openSeparator) && StringUtils.isNotBlank(openSeparator)) {
```

Comment:
Maybe this sould be closeSeparator?

StringUtils.isNoneBlank(openSeparator, closeSeparator) ?

Reply:
Oh, yes! Copypasting is always dangerous...